python/cuml/cuml/tests/test_fil.py

```diff
@@ -19,6 +19,7 @@
 import numpy as np
 import pandas as pd
 import pytest
+import treelite
```

Comment:
This line might be the culprit for the TLS error in libgomp. Treelite (`manylinux2014` wheel) ships with an old GLIBC, which is known to [cause issues with static TLS](https://github.com/microsoft/LightGBM/issues/6509#issuecomment-2196195319). I will try moving this import and see if that fixes the error.